hnn_core/cell.py

```diff
@@ -231,6 +234,34 @@ def insert_dipole(self, yscale):
             dpp.ztan = y_diff[-1]
         self.dipole = h.Vector().record(self.dpl_ref)
 
+    def add_tonic_input(self, amplitude, t0, T, loc=0.5):
+        """Create tonic feed at the soma.
+
+        Parameters
+        ----------
+        amplitude : float
+            The amplitude of the input.
+        t0 : float
+            The start time of tonic input (in ms).
+        T : float
+            The end time of tonic input (in ms).
+        loc : float (0 to 1)
+            The location of the input in the section.
+        """
+        duration = T - t0
+        # names must be actual section names, or else it will fail silently
+        self.tonic_feeds = list()
+        for sect_name in ['soma']:
+            seclist = h.SectionList()
+            seclist.wholetree(sec=self.soma)
```

Comment:
Is there a particular reason why we need a for-loop here? Unless I'm missing something, I feel like this could be a lot simpler.

Reply:
copied from old code. Agree this could be simpler :)

Reply:
done